src/main/kotlin/org/jitsi/jibri/service/impl/FileRecordingJibriService.kt

```diff
@@ -224,9 +234,34 @@ class FileRecordingJibriService(private val fileRecordingParams: FileRecordingPa
                 // Make sure we get all the logs
                 streamDone.get()
                 logger.info("Recording finalize script finished with exit value $exitValue")
+                when (exitValue) {
+                    0 -> handleFinalizeSuccess()
+                    else -> handleFinalizeError()
+                }
             }
         } catch (e: Exception) {
             logger.error("Failed to run finalize script: $e")
         }
     }
+
+    private fun handleFinalizeSuccess() {
+        logger.info("Finalize finished successfully, deleting files")
+        try {
+            Files.walk(sessionRecordingDirectory)
+                // Makes sure we process the directories after the files within them
+                .sorted(Comparator.reverseOrder())
+                .map(Path::toFile)
+                .forEach { f -> f.delete() }
+        } catch (e: Exception) {
+            logger.error("Error deleting session recording directory", e)
+        }
+    }
+
+    private fun handleFinalizeError() {
+        logger.error("Error during finalize script, moving recording to failure directory")
```

Comment:
Does that result in the recorded files being moved to `recordingDirectory/failed/sessionId/sessionId`?

Reply:
Nope, it uses the second argument as the entire target, which I found confusing as well.  I just double-checked if it could be done that way with the `Files.move` but it doesn't work :/